grpc/src/main/java/io/stargate/grpc/service/ValuesHelper.java

```diff
@@ -164,6 +167,56 @@ private static ResultSet processResult(Rows rows, boolean skipMetadata) throws S
     return resultSetBuilder.build();
   }
 
+  public static EnrichedResultSet processEnrichedResult(
+      Rows rows,
+      boolean skipMetadata,
+      BiFunction<List<Column>, io.stargate.db.datastore.Row, ByteBuffer> getComparableBytes,
+      Function<io.stargate.db.datastore.Row, ByteBuffer> getPagingState,
+      BiFunction<List<Column>, List<ByteBuffer>, io.stargate.db.datastore.Row> makeRow)
+      throws StatusException {
+    final List<Column> columns = rows.resultMetadata.columns;
+    final int columnCount = columns.size();
+
+    EnrichedResultSet.Builder resultSetBuilder = EnrichedResultSet.newBuilder();
+    if (!skipMetadata) {
+      for (Column column : columns) {
+        resultSetBuilder.addColumns(
+            ColumnSpec.newBuilder()
+                .setType(convertType(columnTypeNotNull(column)))
+                .setName(column.name())
+                .build());
+      }
+    }
+
+    for (List<ByteBuffer> row : rows.rows) {
+      EnrichedRow.Builder enrichedRowBuilder = EnrichedRow.newBuilder();
+      Row.Builder rowBuilder = Row.newBuilder();
+      io.stargate.db.datastore.Row arrayListRow = makeRow.apply(columns, row);
+      ByteBuffer comparableBytes = getComparableBytes.apply(columns, arrayListRow);
+      ByteBuffer rowPagingState = getPagingState.apply(arrayListRow);
+      for (int i = 0; i < columnCount; ++i) {
+        ColumnType columnType = columnTypeNotNull(columns.get(i));
+        ValueCodec codec = ValueCodecs.get(columnType.rawType());
+        rowBuilder.addValues(decodeValue(codec, row.get(i), columnType));
+      }
```

Comment:
same as in line `148`, extract utility for creating Row from List<ByteBuffer> with given column spec